caravel/assets/javascripts/SqlLab/reducers.js

```diff
@@ -191,8 +190,10 @@ export const sqlLabReducer = function (state, action) {
       return removeFromArr(state, 'alerts', action.alert);
     },
     [actions.REFRESH_QUERIES]() {
-      queries = action.queries
-      return alterInArr(state, 'queries', queries, { state: 'stopped' });
+      const newQueries = Object.assign({}, state['queries'],
+          action.alteredQueries);
+      console.log(newQueries)
+      return Object.assign({}, state, { ['queries']: newQueries });
```

Comment:
ES6 is cool but no need to use the feature here :),  `{ queries: newQueries }` should do the exact same thing